numpy/ma/core.py

```diff
@@ -2445,32 +2444,32 @@ def _recursive_printoption(result, mask, printopt):
 
 # For better or worse, these end in a newline
 _legacy_print_templates = dict(
-    long_std=textwrap.dedent("""\
-        masked_%(name)s(data =
-         %(data)s,
-        %(nlen)s        mask =
-         %(mask)s,
-        %(nlen)s  fill_value = %(fill)s)
-        """),
-    long_flx=textwrap.dedent("""\
-        masked_%(name)s(data =
-         %(data)s,
-        %(nlen)s        mask =
-         %(mask)s,
-        %(nlen)s  fill_value = %(fill)s,
-        %(nlen)s       dtype = %(dtype)s)
-        """),
-    short_std=textwrap.dedent("""\
-        masked_%(name)s(data = %(data)s,
-        %(nlen)s        mask = %(mask)s,
-        %(nlen)s  fill_value = %(fill)s)
-        """),
-    short_flx=textwrap.dedent("""\
-        masked_%(name)s(data = %(data)s,
-        %(nlen)s        mask = %(mask)s,
-        %(nlen)s  fill_value = %(fill)s,
-        %(nlen)s       dtype = %(dtype)s)
-        """)
+    long_std="""\
+masked_%(name)s(data =
+ %(data)s,
+%(nlen)s        mask =
+ %(mask)s,
+%(nlen)s  fill_value = %(fill)s)
+""",
+    long_flx="""\
+masked_%(name)s(data =
+ %(data)s,
+%(nlen)s        mask =
+ %(mask)s,
+%(nlen)s  fill_value = %(fill)s,
+%(nlen)s       dtype = %(dtype)s)
+""",
+    short_std="""\
+masked_%(name)s(data = %(data)s,
+%(nlen)s        mask = %(mask)s,
+%(nlen)s  fill_value = %(fill)s)
+""",
+    short_flx="""\
+masked_%(name)s(data = %(data)s,
+%(nlen)s        mask = %(mask)s,
+%(nlen)s  fill_value = %(fill)s,
+%(nlen)s       dtype = %(dtype)s)
+"""
 )
```

Comment:
An alternative - `_legacy_print_templates` could also be put behind a function with `functools.lru_cache` - probably 99% of users never hit the code path that needs these.

Reply:
For what it's worth, timing the difference between dedenting and not was ~0.1ms here 0.

Phrased more damningly, it was a factor of 100 slowdown, but on an already fast operation.